core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.controller.ts

```diff
@@ -108,34 +212,51 @@ angular.module('oppia').controller('TranslationModalController', [
     };
 
     $scope.suggestTranslatedText = function() {
-      if (!$scope.uploadingTranslation && !$scope.loadingData) {
-        SiteAnalyticsService.registerContributorDashboardSubmitSuggestionEvent(
-          'Translation');
-        $scope.uploadingTranslation = true;
-        var imagesData = ImageLocalStorageService.getStoredImagesData();
-        ImageLocalStorageService.flushStoredImagesData();
-        ContextService.resetImageSaveDestination();
-        TranslateTextService.suggestTranslatedText(
-          $scope.activeWrittenTranslation.html,
-          TranslationLanguageService.getActiveLanguageCode(),
-          imagesData, function() {
-            AlertsService.addSuccessMessage(
-              'Submitted translation for review.');
-            if ($scope.moreAvailable) {
-              var textAndAvailability = (
-                TranslateTextService.getTextToTranslate());
-              $scope.textToTranslate = textAndAvailability.text;
-              $scope.moreAvailable = textAndAvailability.more;
-            }
-            $scope.previousTranslationAvailable = true;
-            $scope.activeWrittenTranslation.html = '';
-            $scope.uploadingTranslation = false;
-          }, () => {
-            $uibModalInstance.close();
-          });
-      }
-      if (!$scope.moreAvailable) {
-        $uibModalInstance.close();
+      $scope.hasImgCopyError = false;
+      $scope.hasImgTextError = false;
+      const originalElements = angular.element(
+        $scope.textToTranslate);
+      const translatedElements = angular.element(
+        $scope.activeWrittenTranslation.html);
+
+      let translationError = $scope.validateImages(
+        originalElements, translatedElements);
+
```

Comment:
Can we add another basic check that makes sure the number of total html elements are the same between the original and translated content?

Reply:
Or more generally, make sure that each html element in the original has a corresponding html element in the translated?

Reply:
Done.